### PR TITLE
Adapt extension to lib-proctorio 2.0

### PR DESCRIPTION
**Pull request by gabrielfs7**
-- This is a WIP -- 

In order to make the extesion support the new version 2.0 of proctorio library some changes were required: 

https://oat-sa.atlassian.net/browse/NEX-924

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -25,11 +25,12 @@
 	"minimum-stability": "dev",
 	"require": {
 		"oat-sa/oatbox-extension-installer": "~1.1||dev-master",
-		"oat-sa/lib-proctorio": "~1.0",
+		"oat-sa/lib-proctorio": "dev-feature/NEX-924",
```
Comment (mike-ionut-mihai-sandu-tao) [inline composer.json:28]: this needs to be changed

File: composer.json [hunk 2]
```diff
@@ -25,11 +25,12 @@
...
 		"oat-sa/extension-tao-delivery": "*",
 		"oat-sa/generis": "*",
-		"oat-sa/tao-core": "*"
-	},
+		"oat-sa/tao-core": "*",
+        "ext-json": "*"
```
Comment (mike-ionut-mihai-sandu-tao) [inline composer.json:32]: please user phpcs psr12 to format this

File: model/ProctorioApiService.php [hunk 3]
```diff
@@ -59,21 +58,47 @@ class ProctorioApiService extends ConfigurableService
...
-            $providerJsonResponse = $this->requestProctorioUrls($deliveryExecution);
-        if ($this->getValidator()->validate($providerJsonResponse)) {
-            $proctorioUrls = ProctorioResponse::fromJson($providerJsonResponse);
-            $this->getStorage()->set($this->getUrlsId($deliveryExecution), $proctorioUrls);
+        try {
+            $response = $this->requestProctorioUrls($deliveryExecution);
+        } catch (Throwable $exception) {
+            /*
+               @TODO @FIXME
+               This log was extracted from the previous Validator. Do we really need to make the extension log aware?
```
Comment (mike-ionut-mihai-sandu-tao) [inline model/ProctorioApiService.php:77]: we need to make the extension log aware, tao is the package but it is composed out of multiple extensions

File: model/ProctorioApiService.php [hunk 4]
```diff
@@ -59,21 +58,47 @@ class ProctorioApiService extends ConfigurableService
...
+            );
+
+            throw $exception;
         }
 
-        return $proctorioUrls;
+        $this->getStorage()
+            ->set(
+                $this->getUrlsId($deliveryExecution),
+                json_encode(
```
Comment (mike-ionut-mihai-sandu-tao) [inline model/ProctorioApiService.php:93]: should the response object have this functionality ?

File: composer.json (hunk 2 above)
Comment (gabrielfs7) [reply composer.json]: done

File: model/ProctorioApiService.php [hunk 5]
```diff
@@ -59,21 +58,42 @@ class ProctorioApiService extends ConfigurableService
     private $proctorioService;
 
     /**
-     * @throws GuzzleException
+     * @param DeliveryExecutionInterface $deliveryExecution
```
Comment (boajer) [inline model/ProctorioApiService.php:61]: Looks like redundent doc parameter
```suggestion
```

File: model/ProctorioApiService.php [hunk 6]
```diff
@@ -59,21 +58,42 @@ class ProctorioApiService extends ConfigurableService
...
+     * @return ProctorioResponse
```
Comment (boajer) [inline model/ProctorioApiService.php:62]: Looks like redundent doc parameter

```suggestion

```

**Comment by boajer** [comment]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Pull request merged**